sqlmesh/core/plan/builder.py

```diff
@@ -425,6 +436,64 @@ def _adjust_new_snapshot_intervals(self) -> None:
             if new.is_forward_only:
                 new.dev_intervals = new.intervals.copy()
 
+    def _check_destructive_changes(
+        self, dag: DAG[SnapshotId], directly_modified: t.Set[SnapshotId]
+    ) -> None:
+        def _raise_or_warn(snapshot: Snapshot) -> None:
+            warning_msg = f"Plan results in a destructive change to forward-only model '{snapshot.name}'s schema."
+            if snapshot.model.on_destructive_change.is_warn:
+                logger.warning(warning_msg)
+                return
+            raise PlanError(
+                f"{warning_msg} To allow this, change the model's `on_destructive_change` setting to `warn` or `ignore` or include it in the plan's `--allow-destructive-model` option."
+            )
+
+        snapshot_is_destructive = {}
+        for s_id in dag:
+            # Only process snapshots that are modified by the plan
+            if s_id.name in self._context_diff.modified_snapshots:
+                snapshot = self._context_diff.snapshots[s_id]
+
+                if snapshot and snapshot.is_model:
+                    # should we raise/warn if this snapshot has/inherits a destructive change?
+                    should_raise_or_warn = (
+                        snapshot.model.forward_only or self._forward_only
```

Comment:
Wouldn't this fail if the snapshot is not a model? We should probably check `needs_destructive_check` first

Reply:
Won't the `if snapshot and snapshot.is_model:` ensure only models hit this?

Reply:
Ah, shoot, missed it. Thanks for checking